### PR TITLE
c2c: set replication_lag metric to 0 before high water mark is set

### LINKED ISSUE #97224: streamingestion: address cross cluster replication dbconsole nits
In the first viewing of the cross cluster replication db console page, I noticed a few issues.

During the initial scan:
image

- [ ] Replication Lag should stay at 0, until a high water mark is recorded. It was initially 0 before the job started, and i'm not sure why it bounced up to 1.75.
- [x] Logical Bytes and SST Bytes graphs have the incorrect units and should be monotonically increasing

After Initial Scan:
image

Jira issue: CRDB-24575

### PR DESCRIPTION
**Pull request by msbutler**
Fixes #97224

Release note: None

### CODE REVIEW

**Comment by msbutler** [comment]
Lag is now set to zero before we reach hwm, but it still looks real high right after the hwm is set. Is this ok? (from a run of `c2c/tpcc/warehouses=500/duration=10/cutover=5`)
image

**Comment by msbutler** [comment]
unrelated flake

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go [hunk 1]
```diff
@@ -467,7 +467,11 @@ func (sf *streamIngestionFrontier) maybeUpdatePartitionProgress() error {
 	sf.metrics.JobProgressUpdates.Inc(1)
 	sf.persistedHighWater = f.Frontier()
 	sf.metrics.FrontierCheckpointSpanCount.Update(int64(len(frontierResolvedSpans)))
-	sf.metrics.FrontierLagSeconds.Update(timeutil.Since(sf.persistedHighWater.GoTime()).Seconds())
+	if sf.persistedHighWater != sf.highWaterAtStart {
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go:470]: Out of curiosity, any reason not to compare sf.persistedHighWater to 0 instead?  Just so we don't have to think about what highWaterAtStart actually means in this context?

**Review by stevendanna — APPROVED** [review]

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go:470]: Ah, good q. Suppose the stream ingestion job is paused after making some progress. Then during the initial scan on resume, I _think_ we want Replication Lag to be 0 and not balloon to some ridiculous number. If we just compared `persistedHighWater==0`, then this metric would blow up on resume. If you agree with my reasoning, I'll add a note to explain why we use `highWaterAtStart`.

Comment (stevendanna) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go:470]: I'm not sure I agree.  Once we have a non-zero watermark, then the lag blowing up on resume is actually correct. That is how far behind you are, anything else is kinda a lie.

Comment (msbutler) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go:470]: Sounds good, I'll use persistedHighWater then.

In this case, we ought to also emit a metric with the latest status of the replication job (or any job in general). That way, if a customer wants to alert on high replication lag, they can filter the alert during an initial scan. I can write up an issue, as this seems like lower priority UX work.

**Comment by msbutler** [comment]
TFTR!

bors r=stevendanna

**Pull request merged**